### DISCUSSION TITLE
observability recommendations for hono

### DISCUSSION THREAD

**Discussion by rawwerks** [honojs/hono#3215] [Q&A]
hi! i'm new to hono and i'm specifically interested in porting some of my python serverless functions (modal) to hono (cloudflare). 

does anyone have any recommendations for observability tools for hono (that are compatible with CloudFlare Workers)?

i would ideally ike to use something that supports opentelemetry: https://www.npmjs.com/package/@opentelemetry/api & https://www.npmjs.com/package/@opentelemetry/sdk-trace-base

it would be really cool if there were automated opentelemetry tracing for hono, for example this is what was built for node: https://github.com/open-telemetry/opentelemetry-js/tree/main/packages/opentelemetry-sdk-trace-node

this looks like a nice work-in-progress for bringing opentelemetry to CF workers: https://github.com/evanderkoogh/otel-cf-workers

**Comment by marcomuser** [comment]
Have you looked into https://baselime.io/? It's cloudflare's opentelemtry compatible observability platform.

**Reply by rawwerks** [reply]
wow - this looks amazing, thanks for sharing!

**Comment by jokester** [comment]
found an article about otel with hono in node.js: https://jeremymorrell.dev/blog/minimal-js-tracing/
